scene/2d/animated_sprite_2d.cpp

```diff
@@ -320,9 +321,15 @@ int AnimatedSprite2D::get_frame() const {
 }
 
 void AnimatedSprite2D::set_speed_scale(double p_speed_scale) {
+	if (speed_scale == p_speed_scale) {
+		return;
+	}
+
 	double elapsed = _get_frame_duration() - timeout;
 
-	speed_scale = MAX(p_speed_scale, 0.0f);
+	speed_scale = p_speed_scale;
+	speed_scale_reversed = (speed_scale < 0.0f);
```

Comment:
I suggest it should be done in `NOTIFICATION_INTERNAL_PROCESS`, not here.

```cpp
case NOTIFICATION_INTERNAL_PROCESS: {
    bool is_backward = (!backward && signbit(speed_scale)) || (backward && !signbit(speed_scale));

```

Reply:
Can't quite do.
https://github.com/godotengine/godot/pull/65148/files/5de67e749c6c3fc42c7ebde56a31049ed2ec23e4#diff-65f9883e255929fb24e1b622b003f3b37b03c9423be8e040f4c3881ef5a164ddR1233-R1242
`play()` uses the same `backwards` to begin the animation from the last frame. I can calculate it every time, but keeping it cached as before is nice, too.